bokeh/document.py

```diff
@@ -523,6 +542,37 @@ def remove_periodic_callback(self, callback):
             Throws an error if the callback wasn't added
 
         '''
+        self.remove_session_callback(callback)
+
+    def add_timeout_callback(self, callback, timeout, id=None):
+        ''' Add callback so it can be invoked on a session periodically accordingly to period.
+
+        NOTE: periodic callbacks can only work within a session. It'll take no effect when bokeh output is html or notebook
+
+        '''
+        # create the new callback object
+        cb = TimeoutCallback(self, callback, timeout, id)
+        self._session_callbacks[callback] = cb
+        # emit event so the session is notified of the new callback
+        self._trigger_on_change(SessionCallbackAdded(self, cb))
+        return cb
+
+    def remove_timeout_callback(self, callback):
+        ''' Remove a callback added earlier with add_timeout_callback()
+
+            Throws an error if the callback wasn't added
+
+        '''
+        self.remove_session_callback(callback)
+
+
+    def remove_session_callback(self, callback):
```

Comment:
@bokeh/core I'm doing this repetition (having different remove for timeout and periodic and a remove session) for the sake expressiveness.. but feel if could be better. Any suggestions? I'm thinking to at least turn `remove_session_callback` private..